Sources/Implementation/DefaultDecisionService.swift

```diff
@@ -190,42 +212,101 @@ class DefaultDecisionService: OPTDecisionService {
+//                                                         user: user,
+//                                                         options: options)
+//        reasons.merge(decisionResponse.reasons)
+//        if let decision = decisionResponse.result {
+//            return DecisionResponse(result: decision, reasons: reasons)
+//        }
+//        
+//        return DecisionResponse(result: nil, reasons: reasons)
+    }
+    
+    func getVariationForFeatureList(config: ProjectConfig,
+                                    featureFlags: [FeatureFlag],
+                                    user: OptimizelyUserContext,
+                                    options: [OptimizelyDecideOption]? = nil) -> [DecisionResponse<FeatureDecision>] {
+        
         let reasons = DecisionReasons(options: options)
+        let userId = user.userId
+        let ignoreUPS = (options ?? []).contains(.ignoreUserProfileService)
+        var profileTracker: UserProfileTracker?
+        if !ignoreUPS {
+            profileTracker = UserProfileTracker(userId: userId, userProfileService: self.userProfileService, logger: self.logger)
+            profileTracker?.loadUserProfile()
+        }
         
-        // Evaluate in this order:
+        var decisions = [DecisionResponse<FeatureDecision>]()
         
-        // 1. Attempt to bucket user into experiment using feature flag.
-        // Check if the feature flag is under an experiment and the the user is bucketed into one of these experiments
-        var decisionResponse = getVariationForFeatureExperiment(config: config,
-                                                                featureFlag: featureFlag,
-                                                                user: user,
-                                                                options: options)
-        reasons.merge(decisionResponse.reasons)
-        if let decision = decisionResponse.result {
-            return DecisionResponse(result: decision, reasons: reasons)
+        for featureFlag in featureFlags {
+            var decisionResponse = getVariationForFeatureExperiment(config: config, featureFlag: featureFlag, user: user, userProfileTracker: profileTracker)
+            
+            reasons.merge(decisionResponse.reasons)
+            
+            if let decision = decisionResponse.result {
+                decisions.append(DecisionResponse(result: decision, reasons: reasons))
+                continue
+            }
+            
+            decisionResponse = getVariationForFeatureRollout(config: config, featureFlag: featureFlag, user: user)
+            
+            reasons.merge(decisionResponse.reasons)
+            
+            if let decision = decisionResponse.result {
+                decisions.append(DecisionResponse(result: decision, reasons: reasons))
+            } else {
+                decisions.append(DecisionResponse(result: nil, reasons: reasons))
+            }
```

Comment:
We need to separate "FeatureExperiment" and "FeatureRollout".
UPS is only for "FeatureExperiment", and we should not save the decisions from "FeatureRollout".

Reply:
If we didn't catch this with unit testing, can we add new cases covering this?

Reply:
We didn't save UPS for `FeatureRollout`. We only passed the UPS tracker for `getVariationForFeatureExperiment`

Reply:
Yeah, i now see that tracker is not used for rollout.